handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -1557,14 +1559,26 @@ public final synchronized void setSSLParameters(SSLParameters sslParameters) {
 
         int version = PlatformDependent.javaVersion();
         if (version >= 7) {
-            endPointIdentificationAlgorithm = sslParameters.getEndpointIdentificationAlgorithm();
+            final String endPointIdentificationAlgorithm = sslParameters.getEndpointIdentificationAlgorithm();
+            final boolean endPointVerificationEnabled = endPointIdentificationAlgorithm != null &&
+                    !endPointIdentificationAlgorithm.isEmpty();
+            SSL.setHostNameValidation(ssl, DEFAULT_HOSTNAME_VERIFICATION_FLAGS,
+                    endPointVerificationEnabled ? getPeerHost() : null);
```

Comment:
@Scottmitch does this mean we need to check if HTTPS / LDAPS is used and if not throw an exception ?